packages/console/src/components/FileUploader/index.tsx

```diff
@@ -0,0 +1,123 @@
+import type { UserAssetsResponse, AllowedUploadMimeType } from '@logto/schemas';
+import { maxUploadFileSize } from '@logto/schemas';
+import classNames from 'classnames';
+import { useCallback, useState } from 'react';
+import { useDropzone } from 'react-dropzone';
+import { useTranslation } from 'react-i18next';
+
+import UploaderIcon from '@/assets/images/upload.svg';
+import useApi from '@/hooks/use-api';
+
+import { Ring } from '../Spinner';
+import * as styles from './index.module.scss';
+
+const allowedFileCount = 1;
+
+type Props = {
+  maxSize: number; // In kilobytes
+  allowedMimeTypes: AllowedUploadMimeType[];
+  limitDescription: string;
+  onCompleted: (fileUrl: string) => void;
+};
+
+const FileUploader = ({ maxSize, allowedMimeTypes, limitDescription, onCompleted }: Props) => {
+  const { t } = useTranslation(undefined, { keyPrefix: 'admin_console' });
+  const [isUploading, setIsUploading] = useState(false);
+  const [uploaderError, setUploaderError] = useState<string>();
+  const hasError = Boolean(uploaderError);
+
+  const api = useApi();
+
+  const onDrop = useCallback(
+    async (acceptedFiles: File[]) => {
+      setUploaderError(undefined);
+
+      if (acceptedFiles.length > allowedFileCount) {
```

Comment:
Should we replace the source directly?

Reply:
It may be confusing when the user drops more than two files.
We will have errors to let the user know what is happening.